src/diagnostics/land_compute_methods.jl

```diff
@@ -0,0 +1,145 @@
+"""
+    @land_compute name model compute
+
+
+Macro generating a function to compute a land diagnostic,
+needed in the ClimaDiagnostics framework.
+
+For example,
+@land_compute "soil_net_radiation" SoilCanopyModel p.soil.R_n
+
+generates the function
+
+function compute_soil_net_radiation!(out, Y, p, t, land_model::SoilCanopyModel)
+    if isnothing(out)
+        return copy(p.soil.R_n)
+    else
+        out .= p.soil.R_n
+    end
+end
+"""
+macro land_compute(name, model, compute)
+    functions = Expr[]
+    function_name = Symbol("compute_", name, "!")
+    function_body = esc(quote
+        function $function_name(out, Y, p, t, land_model::$model)
+            if isnothing(out)
+                return copy($compute)
+            else
+                out .= $compute
+            end
+        end
+    end)
+    push!(functions, function_body)
+    return quote
+        $(functions...)
+    end
+end
+
+### BucketModel ###
+
+# variables stored in p (diagnostics variables stored in the cache)
+@land_compute "aerodynamic_resistance" BucketModel p.bucket.turbulent_fluxes.r_ae
+@land_compute "albedo" BucketModel p.bucket.α_sfc
+@land_compute "latent_heat_flux" BucketModel p.bucket.turbulent_fluxes.lhf
+@land_compute "net_radiation" BucketModel p.bucket.R_n
+@land_compute "sensible_heat_flux" BucketModel p.bucket.turbulent_fluxes.shf
+@land_compute "surface_air_density" BucketModel p.bucket.ρ_sfc
+@land_compute "surface_specific_humidity" BucketModel p.bucket.q_sfc
+@land_compute "surface_temperature" BucketModel p.bucket.T_sfc
+@land_compute "vapor_flux" BucketModel p.bucket.turbulent_fluxes.vapor_flux
```

Comment:
This is also a bit unusual. Water vapor flux? Can we call it water flux? Is it ET?

Reply:
@kmdeck

Reply:
we can call it water vapor flux! I guess for the bucket it is ET. for soil, it is evaporation+sublimation, for canopy, transpiration